OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -209,6 +209,9 @@ public bool HasAnyValidWeapons(Target t, bool checkForCenterTargetingWeapons = f
 			// PERF: Avoid LINQ.
 			foreach (var armament in Armaments)
 			{
+				if (armament.IsTraitDisabled)
+					continue;
```

Comment:
Ok, you are right. After testing you are right this fixes the problem.

In fact i tried to fix that myself too and put the check below as 	

    if (checkIsValid && !armament.IsTraitDisabled && armament.Weapon.IsValidAgainst(t, self.World, self))

so i think that's why i got confused.